plugins/catalog-backend-module-gitea/src/providers/GiteaEntityProvider.ts

```diff
@@ -0,0 +1,266 @@
+    this.config = config;
+    this.integration = integration;
+    this.logger = logger.child({
+      target: this.getProviderName(),
+    });
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+  }
+
+  getProviderName(): string {
+    return `gitea-provider:${this.config.id}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn();
+  }
+
+  private async parseGiteaJsonResponse(
+    response: Response,
+  ): Promise<GiteaProjectQueryResult> {
+    if (!response.ok) {
+      throw new Error(
+        `Failed to fetch Gitea repositories, status: ${response.status}`,
+      );
+    }
+    try {
+      return await response.json();
+    } catch (error) {
+      throw new Error(`Failed to parse Gitea API response: ${error}`);
+    }
+  }
+
+  private createScheduleFn(
+    taskRunner: SchedulerServiceTaskRunner,
+  ): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            class: GiteaEntityProvider.prototype.constructor.name,
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(
+              `${this.getProviderName()} refresh failed, ${error}`,
+              error as Error,
+            );
+          }
+        },
+      });
+    };
+  }
+
+  async refresh(logger: LoggerService): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Gitea discovery connection not initialized');
+    }
+    let allRepos: any[] = [];
+
+    try {
+      // Step 1: Fetch all repos for an organization
+      const OrgRepoApiUrl = `${getGiteaApiUrl(this.integration.config)}orgs/${
+        this.config.organization
+      }/repos`;
+      let page = 1;
+      let hasMoreData = true;
+
+      while (hasMoreData) {
+        const url = `${OrgRepoApiUrl}?page=${page}`;
+        const response = await fetch(url, {
+          method: 'GET',
+          ...getGiteaRequestOptions(this.integration.config),
+        });
+
+        if (!response.ok) {
+          throw new Error(
+            `Failed to fetch page ${page} of repos, status: ${response.status}`,
+          );
+        }
+
+        const projectsPage = (await this.parseGiteaJsonResponse(
+          response,
+        )) as GiteaProjectQueryResult;
+
+        if (!Array.isArray(projectsPage) || projectsPage.length === 0) {
+          hasMoreData = false;
+        } else {
+          allRepos = allRepos.concat(projectsPage);
+          page++;
+        }
+      }
+    } catch (e) {
+      throw new Error(
+        `Failed to list Gitea projects for organization ${this.config.organization}, ${e}`,
+      );
+    }
+
+    // Step 2: Filter for repos that have catalog-info.yaml at root
```

Comment:
```suggestion
    // Filter for repos that have catalog-info.yaml at root
```

You can do that or remove this comment. I don't see a Step 1

Reply:
Oh, there's other numbered comments, maybe do the same? 🤔